### ISSUE TITLE
Fix false positives for trailing combinator in selector-combinator-space-after

### ISSUE THREAD

**Issue by tfrommen** [stylelint/stylelint#4584]
> Clearly describe the bug

This is valid SCSS:

```scss
.foo ~,
.bar {
  .baz {
    display: none;
  }
}
```

It will compile to the following CSS:

```css
.foo ~ .baz,
.bar .baz {
  display: none;
}
```

However, stylelint is complaining about the first selector, because the _general sibling_ combinator, `~`, is not followed by a single whitespace.

> Which rule, if any, is the bug related to?

`selector-combinator-space-after`

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-combinator-space-after": "always"
  }
}
```

> Which version of stylelint are you using?

11.1.1

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

```
stylelint **/*.scss
```

via `@wordpress/scripts`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

No error to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

stylelint is complaining about the first selector, because the `~` combinator is not followed by a single whitespace:

```
Expected single space after "~"   selector-combinator-space-after
```

If I were to add a whitespace (which would look strange anyway), stylelint, of course, complains about a stray whitespace before the comma:

```scss
.foo ~ ,
.bar {
  .baz {
    display: none;
  }
}
```

```
Unexpected whitespace before ","   selector-list-comma-space-before
```

This assumes `"selector-list-comma-space-before": "never"` to be included in the rules.

----

This issue is not specific to the `~` combinator, but all of them. When linting SCSS, stylelint should allow for any combinator to be directly followed by a comma, for example, `~,`.

**Comment by fanich37** [comment]
Thank you for reporting and using the issue template. I can reproduce it on the demo site so it looks like a bug.

**Comment by jeddy3** [comment]
@tfrommen Thanks for the clear explanation.

> I can reproduce it on the demo site so it looks like a bug.

I agree.

The rule should ignore any selectors with a trailing combinator as standard selectors explicitly use the _nesting selector_ (`&`). 

I suggest we add a conditional to [`isStandardSyntaxSelector`](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxSelector.js) to return early if the selector has a trailing [combinator](https://github.com/stylelint/stylelint/blob/master/lib/reference/punctuationSets.js#L7). We'd then make use of that util in [`selectorCombinatorSpaceChecker`](https://github.com/stylelint/stylelint/blob/master/lib/rules/selectorCombinatorSpaceChecker.js).

I've labelled the issue as ready to implement. @tfrommen Please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule) if you have time.

**Comment by malsf21** [comment]
Looks like this is inactive, I should be able to take this!
